### ISSUE TITLE
Push notifications possibly sent twice

### ISSUE THREAD

**Issue by nemhods** [RocketChat/Rocket.Chat#19446]
### Description:

Push notification gateway usage is unexpectedly high, and we suspect double sending of pushes.

### Steps to reproduce:

Unknown

### Expected behavior:

Push notifications are only sent once.

### Actual behavior:

Push notification debug logs show that pushes are sent twice (push IDs anonymized):
```
I20201029-18:00:25.869(0) Push ➔ debug send message "#REDACTED" to userId REDACTED 
I20201029-18:00:25.870(0) Push ➔ debug send to token {   gcm: 'REDACTED1xkfsasd2231' } 
I20201029-18:00:25.871(0) Push ➔ debug send to token {   gcm: 'REDACTED1xkfsasd2231' } 
I20201029-18:00:25.892(0) Push ➔ debug send to token {   apn: 'REDACTED2so2mcxaa2' } 
I20201029-18:00:25.893(0) Push ➔ debug send to token {   apn: 'REDACTED2so2mcxaa2' } 
```

On end devices however, push notifications appear only once.

### Server Setup Information:

  -   Version of Rocket.Chat Server: 3.7.0
  -  Operating System: Ubuntu 20.04
  -  Deployment Method: docker
  -  Number of Running Instances: 1
  -  DB Replicaset Oplog: yes
  -  MongoDB Version: 4.0

### Client Setup Information
Multiple

### Additional context

In light of the upcoming paid push notification change, this is critical for us. Some numbers for background:

* about 200 registered users
* 50 active
* 16000 Push notifications this month
* no bots that send lots of messages
* just two broadcast-channels with sporadic messages

generally, I think these numbers are unreasonable and together with the duplicate logs point to double Push sending.
With our medium usage we shouldn't exceed the maximum purchasable push limit, while our user count is way below the recommended 1000 user maximum shown on the [pricing page](https://rocket.chat/pricing/) for the free edition.

### Relevant logs:

see above.

**Comment by kaiiiiiiiii** [comment]
I see the double amount of notifications in the cloud panel as well. This is a joke! You enforce a 5k limit, but still are not able to provide any real usage data  nor a valid offer to buy more notifications ...

**Comment by Dave-zl** [comment]
We have 55k pushes with little more users, its absolutely impossible...

+ when people are active on desktop, they are getting pushes to mobile phones, why? In the past, it wasnt working this way.

**Comment by zdumitru** [comment]
> We have 55k pushes with little more users, its absolutely impossible...
> 
> * when people are active on desktop, they are getting pushes to mobile phones, why? In the past, it wasnt working this way.

Yes, same here. It is using double the amount of notifications because of that. It used to not send notifications to mobile when the app/web-version is opened.

**Comment by cortex3** [comment]
To add my own experiences here:

We're a company with 100-200 active users per day.
Within those I'm guessing we have maybe ~30 people using the app.
And from my experience I get ~10 push notifications on a WORK day.
So if i calculate 30 apps * 10 messages * 31 days. It would only be around 9300 push notifications.

Yet in september the Graph showed 143.000 push notifications.
This seems impossible to me.

Even if I calculate with all the people online per day (who definitely don't all have apps) It would be 62k per month not 143k.

There seems to be something wrong here.

Something like this to monitor my problem would help:
https://github.com/RocketChat/Rocket.Chat/issues/19482

One Theory is that since this instance is over 4 years old that old push devices do not get removed.
There is already a bug report for this here:
https://github.com/RocketChat/Rocket.Chat/issues/10244

**Comment by OloBo-MSK** [comment]
hello. same problem. any news ?

**Comment by Dave-zl** [comment]
I used 19 823 from limit 10 000 this month, so pushes are not coming anymore :D Situation seems to be "solved" :D

You know, they need to force us to pay, so Rocket did their best in this area.

**Comment by nemhods** [comment]
@OloBo-MSK 
unfortunately not - besides intense looking at migrating to matrix/element, who still provide free unlimited push notifications.
